### PR TITLE
refactor: Kotlin - fix nullability

### PR DESCRIPTION
**Pull request by david-allison**
This is useful before converting 'Utils.java'

**Collection:**

* setUserFlag
* genCards
* cardCount
* emptyCardReport
* updateFieldCache

**BaseSched:**
* remFromDyn

**Anki2Importer:**
* _writeDstMedia

## How Has This Been Tested?

Unit tested

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by lukstbit** [comment]
@david-allison All checks are failing.

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/NoteImporter.kt [hunk 1]
```diff
@@ -11,10 +11,7 @@ import com.ichi2.libanki.*
 import com.ichi2.libanki.template.ParsedNode
 import com.ichi2.libanki.utils.StringUtils
 import com.ichi2.libanki.utils.TimeManager
-import com.ichi2.utils.Assert
-import com.ichi2.utils.HashUtil
-import com.ichi2.utils.HtmlUtils
-import com.ichi2.utils.JSONObject
+import com.ichi2.utils.*
```
Comment (ShridharGoel) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/NoteImporter.kt:14]: Seems like this was done by Android Studio, do you want to revert it?

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/NoteImporter.kt:14]: This is apparently a really contentious area.
I read through a couple threads, and this one (with a focus around here https://github.com/pinterest/ktlint/issues/48#issuecomment-339522957) was informative for me.
It appears that star imports are actually a more Kotlin-y thing (IntelliJ / Android Studio add them by default, the language implementation uses them, and the package is the "unit" of import in Kotlin more than a single file like Java)

So I'm leaning towards not caring about star imports, and ktlint in it's default configuration also does not care, so I'll let this go.

Note that I'm injecting opinion here, it's a matter of taste. Feel free to disagree. If most people disagree, it's possible to configure ktlint to disallow star imports from what I understand, and that's the thing to do so these rules are enforced by tooling not convention

**Review by mikehardy — APPROVED** [review]
LGTM - long comment on star imports which I'm willing to defer to now - it either gets put into our tooling or we allow them (which would be fine by me)

Passes local re-integration

**Pull request merged**